scene/gui/color_picker.cpp

```diff
@@ -1631,8 +1643,13 @@ void ColorPicker::_picker_texture_input(const Ref<InputEvent> &p_event) {
 		if (img.is_valid() && !img->is_empty()) {
 			Vector2 ofs = mev->get_position();
 			picker_color = img->get_pixel(ofs.x, ofs.y);
-			picker_preview_style_box->set_bg_color(picker_color);
-			picker_preview_label->add_theme_color_override(SceneStringName(font_color), picker_color.get_luminance() < 0.5 ? Color(1.0f, 1.0f, 1.0f) : Color(0.0f, 0.0f, 0.0f));
+			picker_preview_style_box_color->set_bg_color(picker_color);
+			picker_preview_style_box->set_bg_color(picker_color.get_luminance() < 0.5 ? Color(1.0f, 1.0f, 1.0f) : Color(0.0f, 0.0f, 0.0f));
+
+			Ref<Image> zoom_previw_img = img->get_region(Rect2i(ofs.x - 8, ofs.y - 8, 17, 17));
+			zoom_previw_img->resize(51, 51, Image::INTERPOLATE_NEAREST);
```

Comment:
You can possibly avoid resizing if you use TextureRect's stretch modes.

Reply:
I do not think there's a way to force `TextureRect` to use nearest neighbor scaling, and zoomed view should have crisp pixels.

Reply:
You can set texture filter (it's CanvasItem property).

Reply:
`picker_texture_zoom->set_texture_filter(CanvasItem::TEXTURE_FILTER_NEAREST);` when it is created.

Reply:
Changed it to `set_texture_filter`.